pkg/components/azure-arc-onboarding/component_test.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2021 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azurearconboarding_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/hcl/v2"
+
+	azurearconboarding "github.com/kinvolk/lokomotive/pkg/components/azure-arc-onboarding"
+	"github.com/kinvolk/lokomotive/pkg/components/util"
+)
+
+func TestConfig(t *testing.T) { //nolint:funlen
+	t.Parallel()
+
+	c := azurearconboarding.NewConfig()
+
+	tests := []struct {
+		desc    string
+		config  string
+		wantErr bool
+	}{
+		{
+			desc: "Valid config",
```

Comment:
I'd use underscores instead of spaces in test description, so we do not sacrifice readability, but it makes it easier to find offending test, as `go test -v` output will convert all spaces in test names to underscores, so you cannot copy-paste-find anymore.
```suggestion
			desc: "Valid_config",
```